### ISSUE TITLE
Create complete list of data to be recorded by intercom per user

### ISSUE THREAD

**Issue by willdoran** [ushahidi/platform-client#132]
Innumerate all pieces of data that should be passed to intercom from the client.

**Comment by jshorland** [comment]
Standard attributes auto gathered by intercom:

- Email (user's email address). 
- Signed up (the day a user first signed up for your product). 
- First seen (the first day a user visited your site or app).
- Last seen (the last day a user visited your site or app). 
- Web sessions (the number of times a user has visited your site or app).
- City and country of user (calculated by their IP address location). 
- Local timezone for user. 
- Last contacted (the date you or a teammate last contacted a user). 
- Last heard from (the last day a user contacted you via message or email). 
- Job title (a user's official job title).
- Social profiles (including Twitter, LinkedIn, Facebook, and more).
- Twitter follower count.
- Device (iOS and Android).
- Operating system.

Custom user data to collect:

- User role assigned to user within deployment (admin, member, etc)
- Deployment URL
- Deployment plan
- $$ Value of plan
- Number of roles in deployment
- Number of users by role in deployment
- Number posts in deployment (both draft & published)
- Data sources configured in deployment
- Number of post types
- Number of notification sign ups (I don't think we can currently collect number of notifications actually SENT, but would be a plus!)
- Number of categories
- Number of saved searches
- Number of collections

Let me know which of these (if any) we would not be able to collect.

**Comment by rjmackay** [comment]
I don't think we can or should collect the counts of posts, etc through the JS client. But I might be able to push them in to the intercom API from our metrics scripts.

**Comment by jshorland** [comment]
We could track these as events rather than data:
- Number of roles in deployment --> creates a role
- Number of users by role in deployment --> creates a user
- Number posts in deployment (both draft & published) --> adds post (could this be for both visitor and user?)
- Data sources configured in deployment --> configures a data source
- Number of post types --> creates post type
- Number of notification sign ups (I don't think we can currently collect number of notifications actually SENT, but would be a plus!) --> signs up for notifications
- Number of categories --> creates a category
- Number of saved searches --> creates a saved search
- Number of collections --> creates a collection, and adds post to collection

Adding: Plan upgrade/downgrade

**Comment by rjmackay** [comment]
+1 on the events. We should add those. Ideally a slightly generic solutions on the tech side as I'd like to push those to more than just Intercom in future.

**Comment by jshorland** [comment]
Important note: this is not for tracking USAGE of the platform. This is so we can send messages to users when they take (or get stuck on) a particular action. This is for enegagement and support, NOT METRICS. we should at no point have to pull usage metrics out of intercom.

**Comment by rjmackay** [comment]
We should also fix the intercom company tracking at the same time

**Comment by rjmackay** [comment]
On tech approach:
- We should take this opportunity to integrate intercom with the API, instead of just frontend.
- Changes look like
    - instrument our use cases so they emit events
    - add event listeners which send data to intercom
    - Add a task to send all existing deployment users to intercom and run it once for all deployments
- The vast majority of these event can be collected in the backend, we should filter out the automated actions however (ie. posts from twitter)
- Deployment plan change etc will need to be tracked on the cloud interface repo
     - We can also try to send org info to intercom too

**Comment by rjmackay** [comment]
@willdoran could you estimate on size?

**Comment by jshorland** [comment]
This is a dependency for implementing this onboarding conept via intercom:
https://docs.google.com/document/d/1oDS_pZrX1TXq21Nqc7nEQRFt_UoeeIzITncmNbfho-o/edit

**Comment by willdoran** [comment]
@jshorland @rjmackay 
For the UX onboarding, the events should live in the client as there will be no API interaction for some of the actions that trigger intercom messages. 
Adding events to the client that trigger automessage via intercom engage is relatively straightforward. I would estimate that it should take about 3 days to add.

For the API, adding the events for creation user/post/roles etc should also be straightforward. This should take an additional 2 days. Will these events also trigger automessages? If not then it would probably be closer to 1.5 days
